scripts/prepare_lima.py

```diff
@@ -0,0 +1,191 @@
+    which stores the preprocessed and tokenized prompts and labels.
+    """
+
+    if hf_access_token is None:
+        raise ValueError(
+            "This dataset requires authentication. Please provide "
+            "your Hugging Face Access Token via `--hf_access_token`."
+        )
+
+    if max_seq_length is None:
+        with open(checkpoint_dir / "lit_config.json", "r", encoding="utf-8") as file:
+            config = json.load(file)
+            max_seq_length = config["block_size"]
+
+    destination_path.mkdir(parents=True, exist_ok=True)
+    print("Loading data file...")
+
+    dataset = load_dataset(data_repo_id, use_auth_token=hf_access_token)
+    train_data = format_dataset(dataset["train"], include_multiturn_conversations)
+
+    # test set is present but doesn't have any solutions, so we cannot use it here
+    # but have to create our own
+    # for consistency with prepare_alpaca.py and prepare_dolly.py
+    # test_set = format_dataset(dataset["test"], include_multiturn_conversations)
+
+    print("Loading tokenizer...")
+    tokenizer = Tokenizer(checkpoint_dir)
+
+    # Partition the dataset into train and test
+    train_set, test_set = random_split(
+        train_data, [1.0 - test_split_fraction, test_split_fraction], generator=torch.Generator().manual_seed(seed)
+    )
+    train_set, test_set = list(train_set), list(test_set)
+
+    print(f"train has {len(train_set):,} samples")
+    print(f"test has {len(test_set):,} samples")
+
+    print("Processing train split ...")
+    train_set = [
+        prepare_sample(
+            example=sample,
+            tokenizer=tokenizer,
+            max_length=max_seq_length,
+            mask_inputs=mask_inputs,
+            ignore_index=ignore_index,
+        )
+        for sample in tqdm(train_set)
+    ]
+    torch.save(train_set, destination_path / "train.pt")
+
+    print("Processing test split ...")
+    test_set = [
+        prepare_sample(
+            example=sample,
+            tokenizer=tokenizer,
+            max_length=max_seq_length,
+            mask_inputs=mask_inputs,
+            ignore_index=ignore_index,
+        )
+        for sample in tqdm(test_set)
+    ]
+    torch.save(test_set, destination_path / "test.pt")
+
+
+def format_dataset(dataset_partition, include_multi_turn_conversations):
+    formatted_ds = []
+
+    if include_multi_turn_conversations:
+        for entry in dataset_partition:
+            convo = entry["conversations"]
+            for i in range(0, len(convo)-1, 2):
+                formatted_ds.append(
+                    {"instruction": convo[i],
+                    "input": "",
+                    "output": convo[i + 1]})
+                
+    else:
+        for entry in dataset_partition:
+            convo = entry["conversations"]
+            formatted_ds.append(
+                {"instruction": convo[0],
+                "input": "",
+                "output": convo[1]})        
```

Comment:
The loop is invariant

```suggestion
    for entry in dataset_partition:
        convo = entry["conversations"]
        if include_multi_turn_conversations:
            for i in range(0, len(convo)-1, 2):
                formatted_ds.append(
                    {"instruction": convo[i],
                    "input": "",
                    "output": convo[i + 1]})
                
        else:
            formatted_ds.append(
                {"instruction": convo[0],
                "input": "",
                "output": convo[1]})      
```

Reply:
Sure, I can do that. I just thought the other way was cheaper (fewer if checks). Thanks!